.githooks/pre-commit

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -euxo pipefail
+
+KT_FILES="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $2 ~ /\.kts|\.kt/ { print $2}')"
+
+if [ -z "$KT_FILES" ];
+then
+  echo "No Kotlin staged files."
+else
+  echo "Running ktlint over these files:"
+  echo "$KT_FILES"
+
+  ./gradlew ktlintFormat -PinternalKtlintGitFilter="$KT_FILES"
+
+  echo "Completed ktlint run."
+
+  echo "Starting detekt run"
+  ./gradlew detekt
+  echo "Completed detekt run"
+
+  echo "$KT_FILES" | while read -r file; do
```

Comment:
Tried 

`for file in "$EDIT_FILES"
do
  if [ -f $file ]; then
    git add $file
  fi
done
`

instead but it didn't work. Since the while loop does what we need I'd propose to just stick with it.

Reply:
As discussed, the while loop is perfectly fine.